src/main/webapp/script.js

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function getLoginLink() {
+  fetch('/login').then(response => response.json()).then(loginInfo => {
+    const loginEl = document.getElementById('login-link');
+    if(loginInfo.status) {
+      document.getElementById('signup-form').classList.remove('hidden');
+      loginEl.innerHTML = 'You are currently logged in with your Google account. Log out <a href=\"' + loginInfo.url + '\">here</a>.';
+    }
+    else {
+      loginEl.innerHTML = 'Login <a href=\"' + loginInfo.url + '\">here</a> to create your account.';
+    }
+  });
+}
+
+function getUserData() {
+  var url = window.location.href;
+  var key = url.split('?')[1];
+
+  fetch('/user?' + key).then(response => response.json()).then(userInfo => {
```

Comment:
Are you aware that this is a security concern?
https://geekflare.com/web-application-injection-attacks/

Reply:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/KeyFactory.html#keyToString-com.google.appengine.api.datastore.Key-
Based on the documentation, I believe including the key in the url is safe.

Reply:
Thanks! I learnt something new today!